Update billable services workspace

agesa3 opened the pull request:
## Requirements

- [ ] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).


## Summary

<!--
Required.
Please describe what problems your PR addresses.
-->


## Screenshots

https://github.com/user-attachments/assets/1f8bf417-3da0-44fa-a872-ecd53d4cb000






*None.*
<!--
Optional.
If possible, please insert any screenshots/videos of your changes here.
Don't forget to remove the *None.* above if you do fill this section.
-->


## Related Issue

*None.*
<!--
Required if applicable.
If present, please link any related issue here, e.g. "https://issues.openmrs.org/browse/123").
Don't forget to remove the *None.* above if you do fill this section.
-->


## Other

*None.*
<!--
Optional.
Anything else that isn't covered by one of the sections above.
Don't forget to remove the *None.* above if you do fill this section.
-->

ojwanganto commented:
@agesa3 could you please attach a video showing what this PR does? Thanks

agesa3 commented:
> @agesa3 could you please attach a video showing what this PR does? Thanks



> @agesa3 could you please attach a video showing what this PR does? Thanks

I've attached a video but still there is an issue with the backend API when it comes to updating I had informed @Injiri about it and he was to action on it since the API creates a new entry rather than updating.If that is sorted out the code will be fully ready since on the frontend everything seems to be ok.

ojwanganto commented:
@agesa3 - You may have forgotten to upload the video. Also, kindly resolve the conficts

agesa3 commented:
> @agesa3 - You may have forgotten to upload the video. Also, kindly resolve the conficts

Actioning on it

ojwanganto commented:
Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.

agesa3 commented:
> Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.

Done.Updated the styling and video

> Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.

Done and updated the video.

@donaldkibet any guidance on this .In my local the import works ok but here on CI it seems the import of WorkspaceContainer is not found.
<img width="1440" alt="Screenshot 2024-07-23 at 11 24 09" src="https://github.com/user-attachments/assets/c34e3b41-9e9b-4ed8-8ae0-1425e1f2d7dc">

donaldkibet commented on packages/esm-billing-app/src/billable-services/billable-services.component.tsx:
```diff
@@ -37,6 +46,12 @@ const BillableServices = () => {
   const [showOverlay, setShowOverlay] = useState(false);
   const [overlayHeader, setOverlayTitle] = useState('');
 
+  const handleEditClick = (initialData) => {
```
Should we have a type for the `initialData` and probably update the name to something close e.g service

donaldkibet reviewed (approved):
Thanks @agesa3

donaldkibet commented:
> > Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.
> 
> Done.Updated the styling and video
> 
> > Thanks, @agesa3. Could you please look at the styling for the edit form? The margins don't look OK.
> 
> Done and updated the video.
> 
> @donaldkibet any guidance on this .In my local the import works ok but here on CI it seems the import of WorkspaceContainer is not found. <img alt="Screenshot 2024-07-23 at 11 24 09" width="1440" src="https://private-user-images.githubusercontent.com/47299413/351246359-c34e3b41-9e9b-4ed8-8ae0-1425e1f2d7dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE3MjQxMjIsIm5iZiI6MTcyMTcyMzgyMiwicGF0aCI6Ii80NzI5OTQxMy8zNTEyNDYzNTktYzM0ZTNiNDEtOWU5Yi00ZWQ4LThhZTAtMTQyNWUxZjJkN2RjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIzVDA4MzcwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1OTM4YmE2MzQ0NTM0MDkxNjNkNTgwZGQzMmM2MTgxMWM4MTZjNjRiMDA1NmY3MzU5NTBmOWM5MDEyZTI4NTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NxXmpX73XflRTlfRKf9E0Obfd1IIiZQGKpv9gCk05E8">

We should update to the latest version of `@openmrs/esm-framework` to fix this error

ojwanganto commented:
@agesa3 - thanks for the updates. You will at least need to refine styling of the form. Kindly ping the team once you have something to be reviewed.

agesa3 commented:
> @agesa3 - thanks for the updates. You will at least need to refine styling of the form. Kindly ping the team once you have something to be reviewed.

@ojwanganto sorry had uploaded the wrong video.Ive corrected and uploaded the correct one

ojwanganto commented:
@agesa3 are you available to complete on this task? We need it urgently

ojwanganto commented:
@agesa3 - so is the problem an issue with the backend?

agesa3 commented:
> @agesa3 - so is the problem an issue with the backend?

Yes, Simon had created a fix and it was merged and it worked ok but then a few days after that it started failing and whenever an update is made it creates a new entry rather than updating.
